### PR TITLE
[Internal] Changes to allow extending the ThroughputProperties class

### PR DESCRIPTION
**Pull request by MehaKaushik**
# Pull Request Template

## Description

This change is needed to allow extending ThroughputProperties class. The need for the same arises because we need to add an additional property on ThroughputProperties. We intend to do this by extending the ThroughputProperties class in Interop layer and to include the additional properties there. 

Since the default constructors are private we are getting blocked. This PR is just changing the same to unblock us. 

The long-term plan for adding this additional property (BackgroundTaskMaxAllowedThroughputPercent) is by making a change in SDK itself. 

## Type of change

- [] New feature (non-breaking change which adds functionality)

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Offer/OfferAutoscaleAutoUpgradeProperties.cs [hunk 1]
```diff
@@ -15,11 +15,11 @@ internal sealed class OfferAutoscaleAutoUpgradeProperties
         /// Default constructor for serialization
         /// </summary>
         [JsonConstructor]
-        private OfferAutoscaleAutoUpgradeProperties()
+        public OfferAutoscaleAutoUpgradeProperties()
         {
         }
 
-        internal OfferAutoscaleAutoUpgradeProperties(int incrementPercent)
+        public OfferAutoscaleAutoUpgradeProperties(int incrementPercent)
```
Comment (MehaKaushik) [inline Microsoft.Azure.Cosmos/src/Resource/Offer/OfferAutoscaleAutoUpgradeProperties.cs:22]: leave this as internal

File: Microsoft.Azure.Cosmos/src/Resource/Settings/ThroughputProperties.cs [hunk 2]
```diff
@@ -33,14 +33,14 @@ public class ThroughputProperties
         /// Default constructor for serialization
         /// </summary>
         [JsonConstructor]
-        private ThroughputProperties()
+        public ThroughputProperties()
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/ThroughputProperties.cs:36]: This is the only public type change right? 
Can internal do?

Comment (MehaKaushik) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/ThroughputProperties.cs:36]: Yes internal would also do.

Comment (MehaKaushik) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/ThroughputProperties.cs]: Updated to internal

**Review by kirankumarkolli — APPROVED** [review]

**Review by vivekr20 — APPROVED** [review]

**Pull request merged**